src/models/Team.php

```diff
@@ -734,29 +766,146 @@ public static function regenerateHash(string $password_hash): bool {
     Control::invalidateMCRecords('ALL_ACTIVITY'); // Invalidate Memcached Control data.
   }
 
-  public static async function genGetLiveSyncKey(
+  public static async function genAuthTokenExists(
+    string $type,
+    string $token,
+  ): Awaitable<bool> {
+    $db = await self::genDb();
+
+    $team_id_result = await $db->queryf(
+      'SELECT team_id FROM teams_oauth WHERE type = %s AND token = %s',
+      $type,
+      $token,
+    );
+
+    if ($team_id_result->numRows() === 1) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  public static async function genTeamOAuthTokenExists(
+    string $type,
     int $team_id,
+  ): Awaitable<bool> {
+    $db = await self::genDb();
+
+    $team_id_result = await $db->queryf(
+      'SELECT id FROM teams_oauth WHERE type = %s AND team_id = %d',
+      $type,
+      $team_id,
+    );
+
+    if ($team_id_result->numRows() === 1) {
```

Comment:
style nit: we'd generally write:

```
return ($team_id_result->numRows() === 1);
```

Also, when if blocks return, we tend to not do an else block, to reduce nesting, and make it clearer what the flow is:

```
if (foo) {
  return true;
}
return false;
```

This applies throughout your changes.

Reply:
This is another one that would be good to fix throughout the code base.